test/cmd/aro-hcp-tests/cleanup.go

```diff
@@ -24,84 +24,286 @@ import (
+			if err := cleanupCmd.Flags().Set("delete-expired", "true"); err != nil {
+				return fmt.Errorf("failed to set delete-expired flag: %w", err)
+			}
+			// set the other flags
+			if dryRun {
+				if err := cleanupCmd.Flags().Set("dry-run", "true"); err != nil {
+					return fmt.Errorf("failed to set dry-run flag: %w", err)
+				}
+			}
+			if nowString != "" {
+				if err := cleanupCmd.Flags().Set("now", nowString); err != nil {
+					return fmt.Errorf("failed to set now flag: %w", err)
+				}
+			}
+
+			// Execute the cleanup command
+			return cleanupCmd.RunE(cleanupCmd, args)
+		},
+	}
+	cmd.Flags().StringVar(&nowString, "now", nowString, "Current time when evaluating expiration (RFC3339)")
+	cmd.Flags().BoolVar(&dryRun, "dry-run", false, "Print which resource groups would be deleted without deleting")
+
+	return cmd
+}
+
+func NewCleanupResourceGroupsCommand() *cobra.Command {
+	const deletionTimeout = 20 * time.Minute
+
+	var resourceGroups []string
+	var deleteExpired bool
+	nowString := time.Now().Format(time.RFC3339)
+	var dryRun bool
+
+	cmd := &cobra.Command{
+		Use:          "cleanup-resource-groups",
+		Short:        "Delete resource groups (explicit list or expired resource groups)",
 		SilenceUsage: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx, cancelCause := context.WithCancelCause(context.Background())
+			ctx, cancelCause := newCleanupContext()
 			defer cancelCause(errors.New("exiting"))
 
-			abortCh := make(chan os.Signal, 2)
-			go func() {
-				<-abortCh
-				fmt.Fprintf(os.Stderr, "Interrupted, terminating tests")
-				cancelCause(errors.New("interrupt received"))
-
-				select {
-				case sig := <-abortCh:
-					fmt.Fprintf(os.Stderr, "Interrupted twice, exiting (%s)", sig)
-					switch sig {
-					case syscall.SIGINT:
-						os.Exit(130)
-					default:
-						os.Exit(130) // if we were interrupted, never return zero.
+			tc := framework.NewTestContext()
+
+			var resourceGroupsToDelete []string
+
+			// If resource groups are explicitly provided, use those
+			if len(resourceGroups) > 0 {
+				resourceGroupsToDelete = resourceGroups
+			} else if deleteExpired {
+				// If no resource groups provided, use deleteExpired logic
+				now, err := time.Parse(time.RFC3339, nowString)
+				if err != nil {
+					return fmt.Errorf("failed to parse --now value: %w", err)
+				}
+
+				expiredResourceGroups, err := framework.ListAllExpiredResourceGroups(
+					ctx,
+					tc.GetARMResourcesClientFactoryOrDie(ctx).NewResourceGroupsClient(),
+					now,
+				)
+				if err != nil {
+					return fmt.Errorf("failed to list expired resource groups: %w", err)
+				}
+
+				resourceGroupsToDelete = make([]string, 0, len(expiredResourceGroups))
+				for _, resourceGroup := range expiredResourceGroups {
+					resourceGroupsToDelete = append(resourceGroupsToDelete, *resourceGroup.Name)
+				}
+			}
+
+			if len(resourceGroupsToDelete) == 0 {
+				fmt.Println("No resource groups provided")
```

Comment:
Any specific reason not to be using the logger provided to commands in this repo otherwise?

Reply:
Added with e8cf9666